app/ui-react/syndesis/src/modules/integrations/components/editor/EditorSteps.tsx

```diff
@@ -0,0 +1,81 @@
+import { isConfigurationRequired } from '@syndesis/api';
+import * as H from '@syndesis/history';
+import {
+  ConnectionCard,
+  ConnectionsGrid,
+  ConnectionsGridCell,
+  ConnectionSkeleton,
+} from '@syndesis/ui';
+import { WithLoader } from '@syndesis/utils';
+import * as React from 'react';
+import { Translation } from 'react-i18next';
+import { ApiError, EntityIcon } from '../../../../shared';
+import { IUIStep } from './interfaces';
+import { isTechPreview } from './utils';
+
+export interface IEditorStepsProps {
+  error: boolean;
+  loading: boolean;
+  steps: IUIStep[];
+
+  getEditorStepHref(editorStep: IUIStep): H.LocationDescriptor;
+}
+
+export class EditorSteps extends React.Component<IEditorStepsProps> {
+  public render() {
+    return (
+      <Translation ns={['connections', 'shared']}>
+        {t => (
+          <ConnectionsGrid>
+            <WithLoader
+              error={this.props.error}
+              loading={this.props.loading}
+              loaderChildren={
+                <>
+                  {new Array(5).fill(0).map((_, index) => (
+                    <ConnectionsGridCell key={index}>
+                      <ConnectionSkeleton />
+                    </ConnectionsGridCell>
+                  ))}
+                </>
+              }
+              errorChildren={<ApiError />}
+            >
+              {() => {
+                return this.props.steps.map((s, index) => {
+                  const configurationRequired =
+                    s.board && isConfigurationRequired(s.board);
```

Comment:
I think this is going to throw for non-connections, since `board` will be `undefined` and `isConfigurationRequired` is assuming board _will_ be defined.